### PR TITLE
feat: Automatically closes the DropdownContainer when scrolling outside of the component

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Automatically closes the `DropdownContainer` when scrolling outside of the component.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/204912730-974ed14f-ec05-443f-b6c2-5236ba12eaaf.mov

### TESTING INSTRUCTIONS
Check the video for instructions.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/DropdownContainer/index.tsx [hunk 1]
```diff
@@ -249,6 +249,14 @@ const DropdownContainer = forwardRef(
       [ref],
     );
 
+    // Closes the popover when scrolling on the document
+    useEffect(() => {
+      document.onscroll = () => setPopoverVisible(false);
```
Comment (kgabryje) [inline superset-frontend/src/components/DropdownContainer/index.tsx:254]: Can we add throttling? Or call it only if popover is open?

Comment (kgabryje) [reply superset-frontend/src/components/DropdownContainer/index.tsx:254]: Or we could even add the scroll listener when popover opens and remove it when popover closes

Comment (michael-s-molina) [reply superset-frontend/src/components/DropdownContainer/index.tsx]: Good suggestion. I changed the code to remove the listener when closing the popover and tested that it does not require throttling.

**Review by kgabryje — APPROVED** [review]
LGTM!

**Pull request merged**